### PR TITLE
Added a simple tracing feature and refactored the tests.

### PR DESCRIPTION
**Pull request by Sqeaky**

### CODE REVIEW

File: include/Trace.h [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#ifndef MEZZ_TRACE // The user macro
+
+    /// @def MEZZ_Trace
+    /// @brief This is a compiler flag to enable or disable the trace messages.
+
+    /// @def MEZZ_TRACE
+    /// @brief This is a better alternative to littering prints in code when troubleshooting without a debugger.
+    /// @details When MEZZ_Trace (Lower case) is enable this is nothing, When enabled this emits a message, file name,
+    /// line number, and function name it is called from.
+    /// @param Message A message to be emitted
```
Comment (MakoEnergy) [inline include/Trace.h:57]: I think this should be "When Mezz_Trace (Lower case) is disabled this is nothing,"

File: include/Trace.h [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+#endif
+
+
+namespace Mezzanine
+{
+namespace StaticFoundation
+{
+
+    /// @brief A simple function that sends the provided information
+    /// @note This shouldn't be used directly by most cost. instead game devs should use MEZZ_TRACE.
```
Comment (MakoEnergy) [inline include/Trace.h:79]: By most cost?

File: include/Trace.h [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    #endif
+#endif
+
+
+namespace Mezzanine
+{
+namespace StaticFoundation
+{
+
+    /// @brief A simple function that sends the provided information
```
Comment (MakoEnergy) [inline include/Trace.h:78]: /nit Sentences end with periods.

File: test/Tests.cpp [hunk 4]
```diff
@@ -146,70 +184,56 @@ NameValuePairMap CheckableValues()
...
-         << "\nCompiled in:\n" << Stringify(Compiled);
+    String Results;
+    for(Mezzanine::NameValuePair pair : Mapping)
+        { Results += "  " + pair.first + ": " + pair.second + "\n"; }
+    return Results;
+}
 
-    stringstream XmlContents;
-    XmlContents << "<testsuite tests=\"" << Expected.size() << "\">\n";
+String IntToString(Mezzanine::Int32 SomeInt)
```
Comment (MakoEnergy) [inline test/Tests.cpp:195]: const?

File: include/Trace.h (hunk 1 above)
Comment (Sqeaky) [reply include/Trace.h:57]: You are correct, I failed my basic boolean logic. This will be fixed in the next commit.

File: include/Trace.h (hunk 2 above)
Comment (Sqeaky) [reply include/Trace.h:79]: I revised this whole block, will be in next commit.

File: include/Trace.h (hunk 3 above)
Comment (Sqeaky) [reply include/Trace.h:78]: I will review all my docs before pushing again.

File: test/Tests.cpp (hunk 4 above)
Comment (Sqeaky) [reply test/Tests.cpp:195]: I didn't change this function other than maybe to move it. I will make this change if it is reasonable.

**Pull request merged**